cachi2/core/models/output.py

```diff
@@ -1,80 +1,80 @@
+log = logging.getLogger(__name__)
 
 
-class _DependencyBase(pydantic.BaseModel):
-    """Common attributes for all dependency types."""
+class Component(pydantic.BaseModel):
+    """A software component such as a dependency or a package.
 
-    type: PackageType
-    name: str
-
-
-class GomodDependency(_DependencyBase):
-    """Metadata about a gomod dependency."""
-
-    type: Literal["gomod"]
-    version: str
-
-
-class GoPackageDependency(_DependencyBase):
-    """Metadata about a go-package dependency.
-
-    Unlike other dependency types, go-package dependencies may come from the standard library,
-    in which case their version is null.
+    Compliant to the CycloneDX specification:
+    https://cyclonedx.org/docs/1.4/json/#components
     """
 
-    type: Literal["go-package"]
+    name: str
     version: Optional[str]
 
+    type: Literal["library"] = "library"
 
-class PipDependency(_DependencyBase):
-    """Metadata about a pip dependency.
+    # This will soon be replaced by the purl
+    def key(self) -> str:
+        """Uniquely identifies a package.
 
-    Pip dependencies have a 'dev' value to indicate whether the dependency is used at runtime
-    or at build time (requirements.txt X requirements-build.txt).
-    """
+        Used mainly for sorting and deduplication.
+        """
+        return f"{self.name}:{self.version}"
 
-    type: Literal["pip"]
-    version: str
-    dev: bool
+    @pydantic.validator("version")
+    def _valid_semver(cls, version: Optional[str]) -> Optional[str]:
+        """Ignore invalid versions.
 
+        For now, versions for the following types of dependencies will be ignored:
+            * Direct files (starting with http|https)
+            * VCS dependencies (starting with git+)
+            * Gomod local replacements (starting with ./)
 
-Dependency = Annotated[
-    Union[GomodDependency, GoPackageDependency, PipDependency],
-    pydantic.Field(discriminator="type"),
-]
+        This behavior is meant to be temporary, proper version output should be handled at package
+        manager level.
+        """
+        regex = re.compile(r"^(git+|https?:\/\/|\./).*$")
 
+        if version is None or regex.match(version):
+            return None
 
-def _dependency_sorting_key(dep: Dependency) -> tuple[str, bool, str, str]:
-    # should return all the attributes a Dependency (the whole Union) can have
-    return dep.type, getattr(dep, "dev", False), dep.name, dep.version or ""
+        return version
 
+    @classmethod
+    def from_package_dict(cls, package: dict[str, Any]) -> "Component":
+        """Create a Component from a Cachi2 package dictionary.
 
-class Package(pydantic.BaseModel):
-    """Metadata about a resolved package and its dependencies."""
+        A Cachi2 package has extra fields which are unnecessary and can cause validation errors.
+        """
+        return Component(name=package.get("name", None), version=package.get("version", None))
```

Comment:
The only reason I added this here is to avoid repeating this pattern several times over the code. But yeah, it is meant to be temporary, it should be removed once we refactor the package manager code.